### ISSUE TITLE
New IPC does not handle debugging gracefully

### ISSUE THREAD

**Issue by colin-grant-work** [eclipse-theia/theia#11203]
### Bug Description:

When debugging in the browser and after sitting on a breakpoint for some time, resuming the application produces tens of thousands of logs and crashes the browser.

### Steps to Reproduce:

0. In the browser application (may also apply to Electron)
1. Set a break point in the developer tools and hit it.
2. Wait a few minutes.
3. Remove your breakpoint and hit continue.
4. Observe tens of thousands of logs about failure to send a message. I took the screenshot below a few seconds into the process - eventually the counter got over 250,000 before the browser killed the page as unresponsive.

@tortmayr, @JonasHelming, @paul-marechal

<img width="624" alt="image" src="https://user-images.githubusercontent.com/62660806/170374360-9b2e45d8-9056-4829-be49-abcad7da1c60.png">

### Additional Information

- Operating System: OS X, Chrome 101
- Theia Version: 9a10b6b5daab5e046f28fde8803b3612e84f6ff4

**Comment by colin-grant-work** [comment]
For me, this is a showstopper - if we don't fix this, we shouldn't release with the new IPC.

**Comment by JonasHelming** [comment]
@colin-grant-work : Just to be sure I get this right: You run a Theia-based application (e.g. the example application) and then you set a break point in this Theia-based application using the developer tools, correct?

**Comment by colin-grant-work** [comment]
> @colin-grant-work : Just to be sure I get this right: You run a Theia-based application (e.g. the example application) and then you set a break point in this Theia-based application using the developer tools, correct?

Exactly. This was in the example application on `master` in the browser, and I set a breakpoint (`@theia/monaco-editor-core/src/vs/editor/common/languages.ts:143`, though I don't think that particular detail matters), browsed some source code for a while, and when I resumed the application after removing my breakpoint, the console flooded and it crashed. I've reproduced it a couple of times.

**Comment by JonasHelming** [comment]
> > @colin-grant-work : Just to be sure I get this right: You run a Theia-based application (e.g. the example application) and then you set a break point in this Theia-based application using the developer tools, correct?
> 
> Exactly. This was in the example application on `master` in the browser, and I set a breakpoint (`@theia/monaco-editor-core/src/vs/editor/common/languages.ts:143`, though I don't think that particular detail matters), browsed some source code for a while, and when I resumed the application after removing my breakpoint, the console flooded and it crashed. I've reproduced it a couple of times.

Might be even related to #11199

**Comment by vince-fugnitto** [comment]
@JonasHelming given how there were already three issues identified after the merge and potentially others perhaps we should revert. The issue with postponing the release is that we cannot identify how long it will take to resolve the bugs and I wouldn’t want to postpone indefinitely. It would have probably been safer overall to have waited for Friday to merge in any case.

**Comment by JonasHelming** [comment]
@vince-fugnitto : I agree and I was actually originally in favor of waiting until Friday. May I ask you or somebody else to do the revert, please? I am traveling without my PC and I do not want to try a revert on my mobile.

**Comment by JonasHelming** [comment]
... and sorry for the trouble

**Comment by colin-grant-work** [comment]
> ... and sorry for the trouble

Not at all. I'm sure it will be a great improvement to the framework, once all the kinks are worked out. Just good to have some time between the merge and the next release to stress test it under a variety of real-world conditions.

**Comment by tortmayr** [comment]
@colin-grant-work Thank you for catching these critical bugs and taking care of the revert. I have now openen a new PR that reapplies the new message-rpc framework and fixes the identfied issues: https://github.com/eclipsesource/theia/pull/40.
